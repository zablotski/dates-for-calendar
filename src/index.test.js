const {
  getCalendarDates,
  getDatesRange,
  getFirstDayOfWeek,
  getLastDayOfWeek,
} = require("./index");

describe("dates for calendar:", () => {
  test("should get calendar dates for given month ", () => {
    const result = getCalendarDates(2023, 0);
    expect(result).toMatchSnapshot();
  });

  test("should get dates from range", () => {
    const result = getDatesRange(
      new Date("2023-01-01"),
      new Date("2023-01-03")
    );
    expect(result).toMatchSnapshot();
  });

  describe("Last day of week: ", () => {
    test("should get last day of week when Monday is first day of week by default", () => {
      const result = getLastDayOfWeek(new Date("2023-01-01"));
      expect(result.toISOString()).toBe("2023-01-01T22:59:59.999Z");
    });

    test("should get last day of week when Monday is first day of week is Sunday", () => {
      const result = getLastDayOfWeek(new Date("2023-01-01"), 0);
      expect(result.toISOString()).toBe("2023-01-07T22:59:59.999Z");
    });

    test("should get last day of week when Monday is first day of week is Saturday", () => {
      const result = getLastDayOfWeek(new Date("2023-01-01"), 6);
      expect(result.toISOString()).toBe("2023-01-06T22:59:59.999Z");
    });
  });

  describe("First day of week: ", () => {
    test("should get last day of week when Monday is first day of week by default", () => {
      const result = getFirstDayOfWeek(new Date("2023-01-01"));
      expect(result.toISOString()).toBe("2022-12-25T23:00:00.000Z");
    });

    test("should get last day of week when Monday is first day of week is Sunday", () => {
      const result = getFirstDayOfWeek(new Date("2023-01-01"), 0);
      expect(result.toISOString()).toBe("2022-12-31T23:00:00.000Z");
    });

    test("should get last day of week when Monday is first day of week is Saturday", () => {
      const result = getFirstDayOfWeek(new Date("2023-01-01"), 6);
      expect(result.toISOString()).toBe("2022-12-30T23:00:00.000Z");
    });
  });
});
