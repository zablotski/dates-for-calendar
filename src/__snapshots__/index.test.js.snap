// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dates for calendar: should get calendar dates for given month  1`] = `
[
  2022-12-25T23:00:00.000Z,
  2022-12-26T23:00:00.000Z,
  2022-12-27T23:00:00.000Z,
  2022-12-28T23:00:00.000Z,
  2022-12-29T23:00:00.000Z,
  2022-12-30T23:00:00.000Z,
  2022-12-31T23:00:00.000Z,
  2023-01-01T23:00:00.000Z,
  2023-01-02T23:00:00.000Z,
  2023-01-03T23:00:00.000Z,
  2023-01-04T23:00:00.000Z,
  2023-01-05T23:00:00.000Z,
  2023-01-06T23:00:00.000Z,
  2023-01-07T23:00:00.000Z,
  2023-01-08T23:00:00.000Z,
  2023-01-09T23:00:00.000Z,
  2023-01-10T23:00:00.000Z,
  2023-01-11T23:00:00.000Z,
  2023-01-12T23:00:00.000Z,
  2023-01-13T23:00:00.000Z,
  2023-01-14T23:00:00.000Z,
  2023-01-15T23:00:00.000Z,
  2023-01-16T23:00:00.000Z,
  2023-01-17T23:00:00.000Z,
  2023-01-18T23:00:00.000Z,
  2023-01-19T23:00:00.000Z,
  2023-01-20T23:00:00.000Z,
  2023-01-21T23:00:00.000Z,
  2023-01-22T23:00:00.000Z,
  2023-01-23T23:00:00.000Z,
  2023-01-24T23:00:00.000Z,
  2023-01-25T23:00:00.000Z,
  2023-01-26T23:00:00.000Z,
  2023-01-27T23:00:00.000Z,
  2023-01-28T23:00:00.000Z,
  2023-01-29T23:00:00.000Z,
  2023-01-30T23:00:00.000Z,
  2023-01-31T23:00:00.000Z,
  2023-02-01T23:00:00.000Z,
  2023-02-02T23:00:00.000Z,
  2023-02-03T23:00:00.000Z,
  2023-02-04T23:00:00.000Z,
]
`;

exports[`dates for calendar: should get dates from range 1`] = `
[
  2023-01-01T00:00:00.000Z,
  2023-01-02T00:00:00.000Z,
  2023-01-03T00:00:00.000Z,
]
`;
